src/cli/parse-flags.ts

```diff
@@ -160,48 +160,61 @@ const configCase = (prop: string) => {
   return prop.charAt(0).toLowerCase() + prop.slice(1);
 };
 
-const ARG_OPTS = {
-  boolean: [
-    'build',
-    'cache',
-    'check-version',
-    'ci',
-    'compare',
-    'debug',
-    'dev',
-    'devtools',
-    'docs',
-    'e2e',
-    'es5',
-    'esm',
-    'headless',
-    'help',
-    'log',
-    'open',
-    'prerender',
-    'prerender-external',
-    'prod',
-    'profile',
-    'service-worker',
-    'screenshot',
-    'serve',
-    'skip-node-check',
-    'spec',
-    'ssr',
-    'stats',
-    'update-screenshot',
-    'verbose',
-    'version',
-    'watch',
-  ],
-  number: ['max-workers', 'port'],
-  string: ['address', 'config', 'docs-json', 'emulate', 'log-level', 'root', 'screenshot-connector'],
-  alias: {
-    config: 'c',
-    help: 'h',
-    port: 'p',
-    version: 'v',
-  },
+const BOOLEAN_ARG_OPTS = [
+  'build',
+  'cache',
+  'check-version',
+  'ci',
+  'compare',
+  'debug',
+  'dev',
+  'devtools',
+  'docs',
+  'e2e',
+  'es5',
+  'esm',
+  'headless',
+  'help',
+  'log',
+  'open',
+  'prerender',
+  'prerender-external',
+  'prod',
+  'profile',
+  'service-worker',
+  'screenshot',
+  'serve',
+  'skip-node-check',
+  'spec',
+  'ssr',
+  'stats',
+  'update-screenshot',
+  'verbose',
+  'version',
+  'watch',
+] as const;
+
+const NUMBER_ARG_OPTS = ['max-workers', 'port'] as const;
+
+const STRING_ARG_OPTS = [
+  'address',
+  'config',
+  'docs-json',
+  'emulate',
+  'log-level',
+  'root',
+  'screenshot-connector',
+] as const;
+
+type AliasMap = Partial<
+  Record<typeof BOOLEAN_ARG_OPTS[number] | typeof NUMBER_ARG_OPTS[number] | typeof STRING_ARG_OPTS[number], string>
+>;
```

Comment:
Minor, we might be able to make this a little easier to understand for folks not used to how indexed types work for arrays here at the cost of a little more verbosity:

```suggestion
type ArrayValuesAsKeys<T extends ReadonlyArray<string>> = T[number];
type OptionKeys =
  | ArrayValuesAsKeys<typeof BOOLEAN_ARG_OPTS>
  | ArrayValuesAsKeys<typeof NUMBER_ARG_OPTS>
  | ArrayValuesAsKeys<typeof STRING_ARG_OPTS>;
type AliasMap = Partial<Record<OptionKeys, string>>;
```

Reply:
yes, that's definitely a little easier to read and figure out!